app/src/org/commcare/tasks/EntityLoaderTask.java

```diff
@@ -50,7 +55,22 @@ public EntityLoaderTask(Detail detail, @Nullable EntityDatum entityDatum, Evalua
     @Override
     protected Pair<List<Entity<TreeReference>>, List<TreeReference>> doInBackground(TreeReference... nodeset) {
         try {
-            return entityLoaderHelper.loadEntities(nodeset[0], this);
+            // Capture sync_case_list_loading trace for performance monitoring
+            Trace trace = CCPerfMonitoring.INSTANCE.createTrace(CCPerfMonitoring.SYNC_ENTITY_LIST_LOADING);
+            if (trace != null && !entityLoaderHelper.isAsyncNodeEntityFactory()) {
+                trace.putAttribute(CCAnalyticsParam.CCHQ_DOMAIN, ReportingUtils.getDomain());
+                trace.putAttribute(CCAnalyticsParam.USERNAME, ReportingUtils.getUser());
+                trace.start();

```

Comment:
can we put all of this in `CCPerfMonitoring. createTrace` instead ? (maybe directly call it 1startTracing`). (the idea being one should not repeat this code for every trace we add)

Reply:
good point, let me do that

Reply:
https://github.com/dimagi/commcare-android/pull/3297/commits/e31df7a3dd2d3e2c4e4176002ef72376daf95aac